### PR TITLE
fix: always set intrinsic gas even if no gas charges

### LINKED ISSUE #10498: (mod-service) always set intrinsic gas for zero-charge `GasCharges`
### Description

A `ContractCallLocal` is not charging intrinsic gas as expected.

### Steps to reproduce

Run `FileUpdateSuite.allUnusedGasIsRefundedIfSoConfigured()`

### Additional context

_No response_

### Hedera network

other

### Version

`develop`

### Operating system

None

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #10498

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCharging.java [hunk 1]
```diff
@@ -96,6 +95,8 @@ public void maybeRefundGiven(
      * Tries to charge gas for the given transaction based on the pre-fetched sender and relayer accounts,
      * within the given context and world updater.
      *
+     * <p>Even if there are gas charges, still returns the intrinsic gas cost of the transaction.
```
Comment (lukelee-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCharging.java:98]: Is this comment accurate?  It implies to me that we will always return intrinsic gas

**Review by lukelee-sl — APPROVED** [review]
LGTM.  Just a question regarding a new comment

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by vtronkov — APPROVED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCharging.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCharging.java:98]: Ugh typo, thanks! Fixing on different branch.

**Pull request merged**